src/app/app-services/courses.service.ts

```diff
@@ -1,83 +1,49 @@
-import { Injectable } from '@angular/core';
-import { todayWithoutTime, yesterdayWithoutTime, tomorrowWithoutTime } from '@app/app-utils/utils';
+import { Injectable, Inject } from '@angular/core';
 import { Course } from '@app/app-models';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CoursesService {
+  constructor(
+    private readonly httpClient: HttpClient,
+    @Inject('BASE_URL') private readonly BASE_URL: string
+  ) { }
 
-  private readonly LOREM_IPSUM = `Lorem ipsum dolor sit amet, 
-  consectetur adipiscing elit, sed do eiusmod tempor incididunt 
-  ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis 
-  nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 
-  consequat. Duis aute irure dolor in reprehenderit in voluptate velit 
-  esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat 
-  cupidatat non proident, sunt in culpa qui officia deserunt mollit anim 
-  id est laborum`;
-
-  private readonly FAKE_COURSE_1 = {
-    id: 'id_1',
-    name: 'Course 1',
-    date: yesterdayWithoutTime(),
-    length: 90,
-    description: `Course 1 description...${this.LOREM_IPSUM}`,
-    isTopRated: false,
-  };
-  private readonly FAKE_COURSE_2 = {
-    id: 'id_2',
-    name: 'Course 2',
-    date: tomorrowWithoutTime(),
-    length: 60,
-    description: `Course 2 description...${this.LOREM_IPSUM}`,
-    isTopRated: false,
-  };
-  private readonly FAKE_COURSE_3 = {
-    id: 'id_3',
-    name: 'Course 3',
-    date: todayWithoutTime(),
-    length: 30,
-    description: `Course 3 description...${this.LOREM_IPSUM}`,
-    isTopRated: true,
-  };
-
-  private courses: Course[] = [];
-
-  constructor() {
-    this.courses = [this.FAKE_COURSE_1, this.FAKE_COURSE_2, this.FAKE_COURSE_3];
-  }
-
-  list(): Course[] {
-    return this.courses;
-  }
-
-  create(course: Course) {
-    this.courses.push(course);
+  list(params: { start?: string, count?: string, textFragment?: string }): Observable<Course[]> {
+    return this.httpClient.get<Course[]>(`${this.BASE_URL}/courses`, { params });
   }
 
-  get(idCourse: string): Course | undefined {
-    const index = this.find(idCourse);
-    if (Number.isInteger(index)) {
-      return this.courses[index];
-    }
+  create(courseInfo: Course): Observable<Course> {
+    return this.httpClient.post<Course>(`${this.BASE_URL}/courses`, {
```

Comment:
As the data types of the `create` method and the API are the same, we may simply pass to `post` the `courseInfo` object instead.

```typescript
return this.httpClient.post<Course>(`${this.BASE_URL}/courses`, courseInfo);
```

Reply:
Done. (Thanks for catching this :smiley: )